crates/core/plugin_sm/src/plugin.rs

```diff
@@ -436,10 +438,29 @@ impl Plugin for ExternalPluginCommand {
         let command = self.command(LIST, None)?;
         let output = self.execute(command, logger).await?;
         if output.status.success() {
-            Ok(deserialize_module_info(
-                self.name.clone(),
-                output.stdout.as_slice(),
-            )?)
+            let config = get_tedge_config().unwrap();
+            let limit = config.query(SoftwarePluginMaxPackagesSetting).unwrap().0;
```

Comment:
This `unwrap()` is safe because `config.query(SoftwarePluginMaxPackagesSetting)` returns always `Ok`. However, it is always recommended to add a comment why it's safe when you use `unwrap()`.

Reply:
I agree with @rina23q comment: using `unwrap()` here is unsafe and also buggy as the config might be stored in a non-default location. The correct way to implement that would be to do the same as for the default plugin type: to let the agent read the config and provide the `TEdgeConfig` value to the plugin. But as this PR is open since quite a long time, I'm okay to have this done in a following PR.